lib/manager/actions/status.js

```diff
@@ -259,6 +260,17 @@ export function handleFinishedJob (job: ServerJob) {
           }
         }
         break
+      case 'PROCESS_FEED':
+        // If viewing a particular feed, navigate to a new feed version as soon as it becomes available.
+        // It is the same path (without version number) as the link shown for this job status.
+        // (If user is not looking at this feed, don't navigate away from their current page.)
+        if (job.feedSourceId) {
+          const newVersionPath = `/feed/${job.feedSourceId}`
+          if (browserHistory.getCurrentLocation().pathname.startsWith(newVersionPath)) {
+            browserHistory.push(newVersionPath)
+          }
+        }
```

Comment:
This generally looks good, a couple of comments:
- there is already logic in the `VALIDATE_FEED` case that fetches the feedSource. I think this check/browser push might be better if it were handled in a `then()` method following that fetch.
- there is a (tiny) bit of nuance missing in this check. If we're already on the feed source path, we don't really need to push to that path (the new version will load automatically). I think what we should be checking here is if someone is viewing a page that startsWith `/feed/${job.feedSourceId}/version`, then we push to `/feed/${job.feedSourceId}`.

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/644/commits/ae230d68aa4b3b1b97d137f9ba63f5a036996e8f. You're right about the check, I intended to do that!